cmd/gravity/src/go/identifier.rs

```diff
@@ -0,0 +1,122 @@
+use std::str::Chars;
+
+use genco::{prelude::*, tokens::ItemStr};
+
+/// Represents a Go identifier with appropriate casing rules.
+///
+/// Go identifiers follow specific naming conventions:
+/// - Public identifiers start with uppercase (exported)
+/// - Private identifiers start with lowercase (unexported)
+/// - Local identifiers are used as-is without transformation
+#[derive(Debug, Clone, Copy)]
+pub enum GoIdentifier<'a> {
+    /// Public/exported identifier (will be converted to UpperCamelCase)
+    Public { name: &'a str },
+    /// Private/unexported identifier (will be converted to lowerCamelCase)
+    Private { name: &'a str },
+    /// Local identifier (will be converted to lowerCamelCase)
+    Local { name: &'a str },
+}
+
+impl<'a> GoIdentifier<'a> {
+    /// Creates a new public identifier.
+    pub fn public(name: &'a str) -> Self {
+        Self::Public { name }
+    }
+
+    /// Creates a new private identifier.
+    pub fn private(name: &'a str) -> Self {
+        Self::Private { name }
+    }
+
+    /// Creates a new local identifier.
+    pub fn local<'b: 'a>(name: &'a str) -> Self {
+        Self::Local { name }
+    }
```

Comment:
Adding these constructors will make it possible to handle the TODO (arcjet/gravity#12) later, but I didn't want to make any behavioral changes in this PR.